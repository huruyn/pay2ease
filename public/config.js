window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";