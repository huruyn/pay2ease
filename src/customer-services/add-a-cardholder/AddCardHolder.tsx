import {
  Col,
  Form,
  Row,
  FormControl,
  Button,
} from 'react-bootstrap';

export default function AddCardHolder() {
  return (
    <div className="page-content">
      <div style={{ padding: 10, backgroundColor: '#F2F3F2' }}>
        <p style={{ textAlign: 'center', marginBottom: 0 }}>Add a Cardholder</p>
      </div>
      <div style={{ padding: 10, backgroundColor: '#FAF0F3' }}>
        <div className="d-flex justify-content-between">
          <p style={{ marginBottom: 0, fontSize: 14 }} className="d-flex">
            <div
              style={{
                width: 20,
                height: 25,
                border: '1px dotted grey',
                textAlign: 'center',
                marginRight: 10,
              }}
            >
              P
            </div>
            We are sorry. You do not have a credit card that allows you to
            complete this request.
          </p>
          <p>
            {'{'}
            Result #0356
            {'}'}
          </p>
        </div>
      </div>
      <div style={{ padding: 10, backgroundColor: '#DEE9F7' }}>
        <p style={{ marginBottom: 0, fontSize: 14 }} className="d-flex">
          <div
            style={{
              width: 20,
              height: 25,
              border: '1px dotted grey',
              textAlign: 'center',
              marginRight: 10,
            }}
          >
            P
          </div>
          <ul>
            <li>
              The additional card will be mailed to the primary cardholder or
              business within 2 or 4 weeks.
            </li>
            <li>All fields are required unless stated as optional.</li>
          </ul>
        </p>
      </div>
      <Row className="mt-4">
        <Col md={12} className="bg-light p-3" style={{ fontWeight: 500 }}>
          <span style={{ color: 'rgb(220, 53, 69)', marginRight: 5 }}>+</span>
          <span>Before you begin</span>
        </Col>
      </Row>
      <div className="mt-3">
        <Row className="mt-3">
          <Col md={6}>
            <Form.Label htmlFor="credit_card">Credit Card</Form.Label>
            <FormControl id="credit_card" aria-describedby="credit_card" />
          </Col>
          <Col md={2}>
            <Form.Label htmlFor="title">Title</Form.Label>
            <FormControl id="title" aria-describedby="title" />
          </Col>
          <Col md={4}>
            <Form.Label htmlFor="first_name">First Name</Form.Label>
            <FormControl id="first_name" aria-describedby="first_name" />
          </Col>
        </Row>
        <Row className="mt-3">
          <Col md={2}>
            <Form.Label htmlFor="credit_card">Initials</Form.Label>
            <FormControl id="credit_card" aria-describedby="credit_card" />
          </Col>
          <Col md={4}>
            <Form.Label htmlFor="title">Last Name</Form.Label>
            <FormControl id="title" aria-describedby="title" />
          </Col>
          <Col md={6}>
            <Form.Label htmlFor="first_name">Date of birth</Form.Label>
            <Row>
              <Col xs="auto">
                <Form.Select>
                  <option>Month</option>
                </Form.Select>
              </Col>
              <Col xs="auto">
                <Form.Select>
                  <option>Day</option>
                </Form.Select>
              </Col>
              <Col xs="auto">
                <Form.Select>
                  <option>Year</option>
                </Form.Select>
              </Col>
            </Row>
          </Col>
        </Row>
        <Row className="mt-3">
          <Col md={6}>
            <Form.Label htmlFor="credit_card">Home phone</Form.Label>
            <FormControl id="credit_card" aria-describedby="credit_card" />
          </Col>
          <Col md={4}>
            <Form.Label htmlFor="title">Business phone</Form.Label>
            <FormControl id="title" aria-describedby="title" />
          </Col>
          <Col md={2}>
            <Form.Label htmlFor="first_name">Extension</Form.Label>
            <FormControl id="first_name" aria-describedby="first_name" />
          </Col>
        </Row>
      </div>
      <div className="mt-4">
        <h5
          style={{
            borderBottom: '1px solid grey',
            paddingBottom: 15,
            paddingTop: 10,
          }}
        >
          Address
        </h5>
        <div className="pt-3 pb-3">
          <Row>
            <Col md={12}>
              <Form.Check
                type="checkbox"
                label="Same address as primary cardholder"
              />
            </Col>
          </Row>
          <Row className="mt-3">
            <Col md={6}>
              <Form.Label htmlFor="address">Address</Form.Label>
              <FormControl id="address" aria-describedby="address" />
            </Col>
            <Col md={6}>
              <Form.Label htmlFor="city">City</Form.Label>
              <FormControl id="city" aria-describedby="city" />
            </Col>
          </Row>
          <Row className="mt-3">
            <Col md={6}>
              <Form.Label htmlFor="province">Province</Form.Label>
              <FormControl id="province" aria-describedby="province" />
            </Col>
            <Col md={6}>
              <Form.Label htmlFor="postal_code">Postal Code</Form.Label>
              <FormControl id="postal_code" aria-describedby="postal_code" />
            </Col>
          </Row>
          <Row className="mt-3">
            <Col md={6}>
              <Form.Label htmlFor="email">Email</Form.Label>
              <FormControl id="email" aria-describedby="email" />
              <Form.Check
                type="checkbox"
                label="Send status updates to this email address"
              />
            </Col>
          </Row>
        </div>
      </div>
      <Row className="mt-4 pb-4" style={{ borderBottom: '1px solid grey' }}>
        <Col md={12} className="bg-light p-3" style={{ fontWeight: 500, borderBottom: '1px solid grey' }}>
          <span style={{ color: 'rgb(220, 53, 69)', marginRight: 5 }}>+</span>
          <span>Terms and Conditions</span>
        </Col>
        <Col md={12} className="mt-3">
          <Form.Check
            type="checkbox"
            label="I have read, understand and agree to the terms and Conditions."
          />
        </Col>
      </Row>
      <div className="mt-4">
        <Row className="mt-4">
          <Col>
            <Button variant="light" className="d-flex">
              <div
                style={{
                  width: 20,
                  height: 25,
                  border: '1px dotted grey',
                  textAlign: 'center',
                  marginRight: 10,
                }}
              >
                P
              </div>
              Cancel
            </Button>
          </Col>
          <Col className="d-flex justify-content-end">
            <Button className="zippy-btn d-flex">
              Next
            </Button>
          </Col>
        </Row>
      </div>
    </div>
  );
}
