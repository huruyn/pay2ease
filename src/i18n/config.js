import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;