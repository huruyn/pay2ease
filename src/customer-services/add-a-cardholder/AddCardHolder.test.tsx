import '../../i18n/config';
import AddCardHolder from './AddCardHolder';

const ReactTestRenderer = require('react-test-renderer');

describe('AddCardHolder Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<AddCardHolder />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
