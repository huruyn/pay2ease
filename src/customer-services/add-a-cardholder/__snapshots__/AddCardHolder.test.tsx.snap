// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardHolder Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    style={
      Object {
        "backgroundColor": "#F2F3F2",
        "padding": 10,
      }
    }
  >
    <p
      style={
        Object {
          "marginBottom": 0,
          "textAlign": "center",
        }
      }
    >
      Add a Cardholder
    </p>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#FAF0F3",
        "padding": 10,
      }
    }
  >
    <div
      className="d-flex justify-content-between"
    >
      <p
        className="d-flex"
        style={
          Object {
            "fontSize": 14,
            "marginBottom": 0,
          }
        }
      >
        <div
          style={
            Object {
              "border": "1px dotted grey",
              "height": 25,
              "marginRight": 10,
              "textAlign": "center",
              "width": 20,
            }
          }
        >
          P
        </div>
        We are sorry. You do not have a credit card that allows you to complete this request.
      </p>
      <p>
        {
        Result #0356
        }
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#DEE9F7",
        "padding": 10,
      }
    }
  >
    <p
      className="d-flex"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 0,
        }
      }
    >
      <div
        style={
          Object {
            "border": "1px dotted grey",
            "height": 25,
            "marginRight": 10,
            "textAlign": "center",
            "width": 20,
          }
        }
      >
        P
      </div>
      <ul>
        <li>
          The additional card will be mailed to the primary cardholder or business within 2 or 4 weeks.
        </li>
        <li>
          All fields are required unless stated as optional.
        </li>
      </ul>
    </p>
  </div>
  <div
    className="mt-4 row"
  >
    <div
      className="bg-light p-3 col-md-12"
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgb(220, 53, 69)",
            "marginRight": 5,
          }
        }
      >
        +
      </span>
      <span>
        Before you begin
      </span>
    </div>
  </div>
  <div
    className="mt-3"
  >
    <div
      className="mt-3 row"
    >
      <div
        className="col-md-6"
      >
        <label
          className="form-label"
          htmlFor="credit_card"
        >
          Credit Card
        </label>
        <input
          aria-describedby="credit_card"
          className="form-control"
          id="credit_card"
        />
      </div>
      <div
        className="col-md-2"
      >
        <label
          className="form-label"
          htmlFor="title"
        >
          Title
        </label>
        <input
          aria-describedby="title"
          className="form-control"
          id="title"
        />
      </div>
      <div
        className="col-md-4"
      >
        <label
          className="form-label"
          htmlFor="first_name"
        >
          First Name
        </label>
        <input
          aria-describedby="first_name"
          className="form-control"
          id="first_name"
        />
      </div>
    </div>
    <div
      className="mt-3 row"
    >
      <div
        className="col-md-2"
      >
        <label
          className="form-label"
          htmlFor="credit_card"
        >
          Initials
        </label>
        <input
          aria-describedby="credit_card"
          className="form-control"
          id="credit_card"
        />
      </div>
      <div
        className="col-md-4"
      >
        <label
          className="form-label"
          htmlFor="title"
        >
          Last Name
        </label>
        <input
          aria-describedby="title"
          className="form-control"
          id="title"
        />
      </div>
      <div
        className="col-md-6"
      >
        <label
          className="form-label"
          htmlFor="first_name"
        >
          Date of birth
        </label>
        <div
          className="row"
        >
          <div
            className="col-auto"
          >
            <select
              className="form-select"
            >
              <option>
                Month
              </option>
            </select>
          </div>
          <div
            className="col-auto"
          >
            <select
              className="form-select"
            >
              <option>
                Day
              </option>
            </select>
          </div>
          <div
            className="col-auto"
          >
            <select
              className="form-select"
            >
              <option>
                Year
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-3 row"
    >
      <div
        className="col-md-6"
      >
        <label
          className="form-label"
          htmlFor="credit_card"
        >
          Home phone
        </label>
        <input
          aria-describedby="credit_card"
          className="form-control"
          id="credit_card"
        />
      </div>
      <div
        className="col-md-4"
      >
        <label
          className="form-label"
          htmlFor="title"
        >
          Business phone
        </label>
        <input
          aria-describedby="title"
          className="form-control"
          id="title"
        />
      </div>
      <div
        className="col-md-2"
      >
        <label
          className="form-label"
          htmlFor="first_name"
        >
          Extension
        </label>
        <input
          aria-describedby="first_name"
          className="form-control"
          id="first_name"
        />
      </div>
    </div>
  </div>
  <div
    className="mt-4"
  >
    <h5
      style={
        Object {
          "borderBottom": "1px solid grey",
          "paddingBottom": 15,
          "paddingTop": 10,
        }
      }
    >
      Address
    </h5>
    <div
      className="pt-3 pb-3"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              disabled={false}
              type="checkbox"
            />
            <label
              className="form-check-label"
              title=""
            >
              Same address as primary cardholder
            </label>
          </div>
        </div>
      </div>
      <div
        className="mt-3 row"
      >
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="address"
          >
            Address
          </label>
          <input
            aria-describedby="address"
            className="form-control"
            id="address"
          />
        </div>
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="city"
          >
            City
          </label>
          <input
            aria-describedby="city"
            className="form-control"
            id="city"
          />
        </div>
      </div>
      <div
        className="mt-3 row"
      >
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="province"
          >
            Province
          </label>
          <input
            aria-describedby="province"
            className="form-control"
            id="province"
          />
        </div>
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="postal_code"
          >
            Postal Code
          </label>
          <input
            aria-describedby="postal_code"
            className="form-control"
            id="postal_code"
          />
        </div>
      </div>
      <div
        className="mt-3 row"
      >
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
            htmlFor="email"
          >
            Email
          </label>
          <input
            aria-describedby="email"
            className="form-control"
            id="email"
          />
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              disabled={false}
              type="checkbox"
            />
            <label
              className="form-check-label"
              title=""
            >
              Send status updates to this email address
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-4 pb-4 row"
    style={
      Object {
        "borderBottom": "1px solid grey",
      }
    }
  >
    <div
      className="bg-light p-3 col-md-12"
      style={
        Object {
          "borderBottom": "1px solid grey",
          "fontWeight": 500,
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgb(220, 53, 69)",
            "marginRight": 5,
          }
        }
      >
        +
      </span>
      <span>
        Terms and Conditions
      </span>
    </div>
    <div
      className="mt-3 col-md-12"
    >
      <div
        className="form-check"
      >
        <input
          className="form-check-input"
          disabled={false}
          type="checkbox"
        />
        <label
          className="form-check-label"
          title=""
        >
          I have read, understand and agree to the terms and Conditions.
        </label>
      </div>
    </div>
  </div>
  <div
    className="mt-4"
  >
    <div
      className="mt-4 row"
    >
      <div
        className="col"
      >
        <button
          className="d-flex btn btn-light"
          disabled={false}
          type="button"
        >
          <div
            style={
              Object {
                "border": "1px dotted grey",
                "height": 25,
                "marginRight": 10,
                "textAlign": "center",
                "width": 20,
              }
            }
          >
            P
          </div>
          Cancel
        </button>
      </div>
      <div
        className="d-flex justify-content-end col"
      >
        <button
          className="zippy-btn d-flex btn btn-primary"
          disabled={false}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
